app/models/basket.rb

```diff
@@ -1,7 +1,32 @@
 class Basket < ApplicationRecord
   has_many :goods, dependent: :destroy
 
+  before_save :update_taxes_update_price
+
   def handle_file(file)
     file.read.split("\n")
   end
+
+  private
+
+  def update_taxes_update_price
+    self.sales_taxes = sum_tax
+    self.total = sum_total
+  end
+
+  def sum_tax
```

Comment:
This is just a matter of style, but you can TDD this by first writing the spec with the different values you expect, and then write the method and run the specs as you write it. Again, this is not wrong, just a preference that you could try and decide what works best for you 😄